src/components/SearchBar/SearchBar.tsx

```diff
@@ -2,24 +2,21 @@ import React, { ChangeEventHandler, MouseEventHandler } from 'react';
 import { FormControl, Typography, InputAdornment } from '@mui/material';
 import TextField from '../TextField';
 import Button from '../Button';
-import {
-  ComponentSize,
-  ComponentVariant,
-  ComponentColor,
-} from '../../config/sizes';
+import { ComponentVariant, ComponentColor } from '../../config/sizes';
 
 export interface SearchBarProps {
-  size?: ComponentSize;
   width?: string;
   value?: string;
-  onChange?: ChangeEventHandler<HTMLInputElement>;
-  onClick?: MouseEventHandler<HTMLButtonElement>;
+  onChange?: (newValue: string) => void;
+  onSearch?: () => void;
```

Comment:
Renamed and changed types. I think the parent component doesn't want to deal with HTML elements or events.